src/test/scala/pl/project13/kaze/KazeClassSpec.scala

```diff
@@ -152,12 +159,113 @@ class KazeClassSpec extends WordSpec with Matchers {
            |  )
            |}""".stripMargin.replaceAll("```", "\"\"\"").split("\n")
 
-      info("Rendered: \n" + rendered)
+      assertEquals(expected, rendered)
+    }
+
+    "create config helpers" in {
+      val rendered: String = KazeClass.of[Simple].withConfig.render
+
+      val expected =
+        """import scala.collection.immutable
+          |import scala.collection.JavaConverters._
+          |import scala.compat.java8.OptionConverters._
+          |import akka.util.JavaDurationConverters._
+          |
+          |final class Simple private(
+          |  val number: Int,
+          |  val text: String,
+          |  val timeout: scala.concurrent.duration.FiniteDuration
+          |) {
+          |
+          |  /** Java API */
+          |  def getNumber: Int = number
+          |  /** Java API */
+          |  def getText: String = text
+          |  /** Java API */
+          |  def getTimeout: java.time.Duration = timeout.asJava
+          |
+          |  def withNumber(value: Int): Simple = copy(number = value)
+          |  def withText(value: String): Simple = copy(text = value)
+          |  /** Scala API */
+          |  def withTimeout(value: scala.concurrent.duration.FiniteDuration): Simple = copy(timeout = value)
+          |  /** Java API */
+          |  def withTimeout(value: java.time.Duration): Simple = copy(timeout = value.asScala)
+          |
+          |  private def copy(
+          |    number: Int = number,
+          |    text: String = text,
+          |    timeout: scala.concurrent.duration.FiniteDuration = timeout
+          |  ): Simple = new Simple(
+          |      number = number,
+          |      text = text,
+          |      timeout = timeout
+          |    )
+          |
+          |  override def toString =
+          |    s```Simple(number=$number,text=$text,timeout=$timeout)```
+          |}
+          |
+          |object Simple {
+          |  /** Scala API */
+          |  def apply(): Simple = new Simple()
+          |  /** Java API */
+          |  def create(): Simple = apply()
+          |  /**
+          |   * Reads from the given config.
+          |   */
+          |  def apply(c: Config): Simple = {
+          |    val number = c.getInt("number")
+          |    val text = c.getString("text")
+          |    val timeout = c.getDuration("timeout").asScala
+          |    apply(
```

Comment:
I'd prefer `new Simple(` because I don't think one would use both config and full apply with parameters.
We typically promote to create an instance from config and then amend it with `.withX`